### DISCUSSION TITLE
what is the difference between train_dreambooth and train_dreambooth_lora?

### DISCUSSION THREAD

**Discussion by Podidiving** [huggingface/diffusers#8840] [Q&A] [answered]
Hi! :wave:

I was looking into examples and got stuck with a question:

there are 2 scripts: [train_dreambooth.py](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth.py) and [train_dreambooth_lora.py](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora.py)
What is the difference between these 2?

As I understood, `train_dreambooth_lora` just trains small lora adapter instead of full unet checkpoint. But then I got puzzled: if that is the case than what is the difference between [train_dreambooth_lora.py](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora.py) and [train_text_to_image_lora.py](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image_lora.py) ?

What am I missing? Can somebody help?

Thanks in advance ❤️

**Comment by asomoza** [comment, marked as answer]
you're almost correct in your assumption, `train_dreambooth_lora` trains a small **dreambooth** lora. 

The difference between `train_dreambooth_lora.py` and `train_text_to_image_lora.py` is that the first one uses [dreambooth](https://dreambooth.github.io/) and the second is just a finetune lora.

**Reply by Podidiving** [reply]
>  trains a small dreambooth lora.

this confuses me a bit.
- `train_dreambooth_lora` trains only lora adapter weights + unique identifier, but uses dreambooth training scheme (i.e. uses prior preserving loss).
- `train_text_to_image_lora`  trains only lora adapter weights using standard training scheme.
-  `train_dreambooth` trains a full model + identifier using dreambooth training scheme

Do I understand it correctly?

**Reply by asomoza** [reply]
yes, that's correct. Some people say that training the full model with dreambooth it's better than a lora, also it allows to later extract different loras with different ranks.

**Reply by Podidiving** [reply]
okay, now it's clear. I was under the impression that when people say about dreambooth they refer to the full-model training, but it's not always the case and it can be referred to the training strategy

thanks again!
